enumerable.rb

```diff
@@ -0,0 +1,96 @@
+ 
+
+module Enumerable
+
+    def my_each
+        for x in 0...self.length
+            yield(self[x])
+            x += 1
+        end
+    end
+
+    def my_each_indexs
+        for x in 0...self.length
+            yield(self[x],x)
+            x += 1
+        end
+    end
+
+    def my_select
+        select_arr = []
+        self.my_each{|x| select_arr << x if yield(x)}
+        select_arr
+    end
+
+    def my_all?
+        initial = false
+        self.my_each {|elem| yield(elem) ? initial = true : initial = false }
+        initial
+    end
+    
+    def my_any?
+        self.my_each do |e|
+          return true if yield(e)
+        end
+        true
+    end
+    
+
+    def my_none?
+        num = true
+        self.my_each{ |x| num = false if yield(x) }
+        num
+    end
+
+    def my_count(obj = nil)
```

Comment:
This method is also not working properly, it return whatever number you pass on if you give it as an argument. Also it does not count the number of elements that return a `true` value if a block is given. Take a close look at the way you wrote your method. The line 49 is forcing a return whenever the count is equal to the argument passed, which is not how it suppose to work. And the same goes for line 50 is just returning whatever count is and not counting the elements.

Reply:
Thank you!!!